### ISSUE TITLE
Update package sets

### ISSUE THREAD

**Issue by thought2** [purescript/registry#339]
```json
{
  "packages": {
    "chameleon-styled": "0.1.0"
  }
}
```

**Comment by thomashoneyman** [comment]
@thought2 Would you mind trying to batch these updates together in the future just to keep the overall number of package sets in check? Thanks for keeping them updated!

**Comment by thought2** [comment]
@thomashoneyman 
sure, actually I tried to. But the following workflow did not work:

Say I have package A and B, whereas B depends on A. Both are in the registry and in the package set.

I am using spago@next in both packages.

Now I need to update package A. I made the changes and published successfully a new version via the spago@next CLI.

Due to that change I need to update package B, too. In my local checkout of package I am adding the new version of package A as an extra dependency in the `spago.yaml` file. With this I can now update package B and run it locally. But when I tried to publish a new version of package B, `spago` tells me that I cannot, since the needed version is not in the package set. I tried with a commit hash or a version as ref.

So the only way out I found in this situation was publish a new version of the package set containing the updated package A in order to publish package B.


The workflows are quite new, so maybe I am overseeing something here. Or is it a spago issue?

**Comment by thomashoneyman** [comment]
cc: @f-f

You should definitely be able to publish packages without all dependencies being in the package sets. This sounds like an issue with Spago.

**Comment by thought2** [comment]
@f-f I could image that cause for this is the same as for: https://github.com/purescript/spago/issues/978
I think the globs are correct but spago just does not download versions (if not in the package set).

**Comment by f-f** [comment]
Yes that's my suspicion as well! I'm trying to replicate locally but didn't manage yet (I'm on holiday this week and only have little screen time)

**Comment by f-f** [comment]
Thanks! I just found the bug 😄

Patch incoming..

**Comment by thought2** [comment]
@f-f I tried to publish with spago `0.93.10` a package that depends on a package version which is not in the package set.

Link to spago.yaml: https://github.com/thought2/purescript-interactive-data.all/blob/50f0de393a657baae3cd62f8e6de95730616b56a/spago.yaml

Package depends on ` - chameleon-styled: ">=1.0.0 <2.0.0"`
1.0.0 is in the registry.
But only 0.1.0 is in the package set.

I am still getting this error:
`❌ Could not find a suitable build plan, the following packages do not point to registry versions: ["chameleon-styled"]`


I removed `.spago` dir before trying.

**Comment by thought2** [comment]
Publishing via GH issue worked

**Comment by f-f** [comment]
@thought2 this issue is unrelated, a few notes:
- when importing an `extra_package` that has a `spago.yaml`, you do not need to specify the `dependencies` as Spago will read them from the imported yaml
- Spago won't let you publish a package if your dependencies point to repos outside the Registry. In this case `chameleon-styled` is pointing to a github repo, but it should point to the registry package, like so:
  ```yaml
  extra_packages:
    chameleon-styled: 1.0.0
  ```

**Comment by thought2** [comment]
@f-f 
- Point one is good news that this is possible now!
- Point two, yes! I forgot about this.

I'll let you know if it worked out when I publish the next time.

**Comment by f-f** [comment]
Thanks! I'll try to clarify that error message to clarify more what's going on

**Comment by thought2** [comment]
@f-f 
Now I tried to publish the next version with a now hopefully correct spago.yaml file:
https://github.com/thought2/purescript-interactive-data.all/blob/40edf5e9103db434f80a8ad72414f783da4fec96/spago.yaml

I am getting a different error now. Below are the full logs (spago `0.93.10`).
Publishing via GH issue worked again.

```
spago publish
[...]

✅ Build succeeded.

Looking for unused and undeclared transitive dependencies...

✅ Ready for publishing. Calling the registry..


✅ Registry accepted the Publish request and is processing...

Logs from the Registry pipeline:
  2023-08-10T05:49:48.686Z Received Publish request, job id: 14583765-67aa-42f0-ade0-3d8c5e461cd7
  2023-08-10T05:49:49.055Z Fetching {
  "githubOwner": "thought2",
  "githubRepo": "purescript-interactive-data.all"
}
  2023-08-10T05:49:49.633Z [NOTIFY] Package source does not have a purs.json file, creating one from your spago.yaml file...
  2023-08-10T05:49:49.697Z Manifest does not typecheck: 

↳ /types/v1/Manifest.dhall

Error: Missing file /types/v1/Manifest.dhall

1│ /types/v1/Manifest.dhall


SCHEMA:1:1
  2023-08-10T05:49:49.701Z Found a valid purs.json file in the package source, but it does not typecheck.

❌ Registry finished processing the package, but it failed. Please fix it and try again.
```

**Comment by f-f** [comment]
Ah, this is a problem in the deployment of the Registry server, cc @thomashoneyman
